src/components/Images/Images.js

```diff
@@ -1,30 +1,150 @@
+  Media.BREAKPOINTS.LAPTOP_RANGE_START,
+  Media.BREAKPOINTS.TABLET_RANGE_START,
+  Media.BREAKPOINTS.PHONE_RANGE_END,
+]
+
+export const CloudinaryImage = ({
+  publicId,
+  className,
+  alt,
+  width,
+  height,
+  crop,
+  ...rest
+}) => {
+
+  // Verify that all required props were supplied
+  const [includesRequired] = useRequired(['publicId', 'alt'])
+  const allRelevantProps = Object.assign({}, rest, {
+    publicId,
+    alt,
+    width,
+    height,
+  })
+  includesRequired(allRelevantProps)
+
+  // Verify that no invalid props were supplied
+  const [includesInvalid] = useInvalid(
+    Object.keys(CloudinaryImage.PUBLIC_PROPS)
+  )
+  includesInvalid(rest)
+
+  const { WEBP, JP2, JPEG, SVG } = CloudinaryImage.IMAGE_FILE_TYPES
+  const baseImageSettings = {
+    quality: 'auto:eco',
+    crop: crop,
+    secure: true,
+    flags: ['progressive:semi'],
+  }
+  let imageClasses = ['Image lazyload', className]
+  let reverseWidth, reverseHeight
+  width && (reverseWidth = width.slice().reverse())
+  height && (reverseHeight = height.slice().reverse())
+
+  const buildImageTag = () => {
+    const format = fileFormats.slice(-1)[0]
+    const imageSettings = {
+      ...baseImageSettings,
+      format: format,
+      dpr: '1.0',
+    }
+
+    return(
+      <img
+        key={`${uuidv4()}`}
+        data-src={cld.url(filePath(publicId),imageSettings)}
+        className={imageClasses.join(' ')}
+        alt={alt}
+      />
+    )
+  }
+
+  const buildSrcTags = (format) => {
+    const dprSettings = ['1.0', '2.0', '3.0']
+    let sourceTags = []
+
+    // We are expecting width/height attribute arrays in the order of
+    // Phone/Tablet/Laptop/Desktop but we have to setup media queries
+    // in the opposite order, so we reverse the arrays here.
+    
+    
+    for (
+      let breakpoint = 0;
+      breakpoint < mediaBreakpoints.length;
+      breakpoint++
+      ) {
 
-const publicIdBase = `https://res.cloudinary.com/${CLOUDINARY_CLOUD_NAME}/image/upload/`
+      const srcsetData = dprSettings.map((dpr, indx)=>{
+        const imageSettings = {
+          ...baseImageSettings,
+          ...(reverseWidth && !!reverseWidth[breakpoint] && { width: reverseWidth[breakpoint] }),
+          ...(reverseHeight && !!reverseHeight[breakpoint] && { height: reverseHeight[breakpoint] }),
+          format: format,
+          dpr: dpr,
+        }
 
-export const CloudinaryImage = ({ publicId, className, crop, ...rest }) => {
-  const publicIdFilename = publicId.replace(publicIdBase, '')
+        return cld.url(filePath(publicId), imageSettings) + ` ${indx + 1}x`
+      })
+
+      const minMax = breakpoint < mediaBreakpoints.length - 1 ? `min` : `max`
+      sourceTags.push(<source
+        key={`${uuidv4()}`}
+        media={`(${minMax}-width: ${mediaBreakpoints[breakpoint]}px)`}
+        data-srcset= {srcsetData.join(', ')}
+        type={`image/${format}`}
+      />)
+      
+    }
+    return sourceTags
+  }
+
+  const isSvg = (publicId) => {
```

Comment:
Non Blocking: Do we think SVG specific stuff should be pushed into it's own custom hook or module? If so, I think we should do it as a follow up task and not block PR

Reply:
Yeah I would vote for not making this support SVGs unless there is a specific reason. The motivator for Cloudinary is the difficulty of handling/serving large raster files basically.